### PR TITLE
feat: initial implementation of OPA subsystem chart and policy catalogue

### PR DESCRIPTION
**Pull request by strieflin**

### CODE REVIEW

**Comment by strieflin** [comment]
The following things have to be done before this PR can be merged:

- [x] Add a Java client implementation that interacts with the policy catalogue service (see #20) 
- [x] Fullfil license obligations
- [x] Check whether all required secrets are available for the workflows

File: services/pkg/catalogue/opa.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	policies = make(map[string]Policy)
+	for _, policy := range result.Policies {
+		policies[policy.ID] = policy
+	}
+	return policies, nil
+}
+
+// fetchPolicyFromOPA fetches a single policy from the OPA service by its ID.
+// Note that there is no way to fetch a single policy by ID, so we fetch all.
```
Comment (sbckr) [inline services/pkg/catalogue/opa.go]: but one could simply call `v1/policies/<POLICY_ID>`

Comment (sbckr) [reply services/pkg/catalogue/opa.go]: see [get-a-policy](https://www.openpolicyagent.org/docs/latest/rest-api/#get-a-policy)

**Review by sbckr — CHANGES_REQUESTED** [review]

File: services/pkg/catalogue/server.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		return
+	}
+	if policy == nil {
+		c.JSON(http.StatusNotFound, gin.H{"error": "policy not found"})
+		return
+	}
+
+	switch c.GetHeader("Accept") {
+	case "application/json":
+		c.JSON(http.StatusOK, policy)
```
Comment (sbckr) [inline services/pkg/catalogue/server.go:73]: why are ID's encoded when listing, but not when fetching individually. That's kinda inconsistent

File: services/pkg/catalogue/opa.go (hunk 1 above)
Comment (strieflin) [reply services/pkg/catalogue/opa.go]: Don't know how I have missed that. Used now.

File: services/pkg/catalogue/server.go (hunk 2 above)
Comment (strieflin) [reply services/pkg/catalogue/server.go:73]: Correct. The ID is now consistently encoded.

**Review by sbckr — DISMISSED** [review]
LGTM

**Review by sbckr — APPROVED** [review]
LGTM

**Pull request merged**